lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -130,6 +130,23 @@ def self.validate_mof_type(mof_type, embeddedinstance_name, name, value)
           end
         end
       end
+
+      # if an edge already exists from Reboot[dsc_reboot] to this resource
+      # Puppet will recognize the cyclic dependency automatically and fail hard:
+      # Error: Failed to apply catalog: Found 1 dependency cycle:
+      # (Dsc_file[foo] => Reboot[dsc_reboot] => Dsc_file[foo])
+      def self.reboot_relationship(resource)
+        reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+
+        # do nothing if no Reboot[dsc_reboot] or already an edge from resource to it
+        return [] if (!reboot_resource ||
+          resource.catalog.relationship_graph.edge?(resource, reboot_resource))
```

Comment:
Thanks to @cowofevil we've discovered this edge check is likely ineffectual / irrelevant at this point in time.  This call should accept the list of other relationships created by `builddepends` and search that instead.  Whoops